rayon-core/src/scope/mod.rs

```diff
@@ -268,20 +275,13 @@ impl<'scope> Scope<'scope> {
     /// own reference to `self` as argument. This can be used to
     /// inject new jobs into `self`.
     pub fn spawn<BODY>(&self, body: BODY)
-        where BODY: FnOnce(&Scope<'scope>) + 'scope
+        where BODY: FnOnce(&Scope<'scope>) + Send + 'scope
     {
         unsafe {
             self.job_completed_latch.increment();
             let job_ref = Box::new(HeapJob::new(move || self.execute_job(body)))
                 .as_job_ref();
-            let worker_thread = WorkerThread::current();
-
-            // the `Scope` is not send or sync, and we only give out
-            // pointers to it from within a worker thread
-            debug_assert!(!WorkerThread::current().is_null());
-
-            let worker_thread = &*worker_thread;
-            worker_thread.push(job_ref);
+            self.registry.inject_or_push(job_ref);
```

Comment:
Let's add a comment here saying that it would be unsafe to just invoke `push` before we don't know that the scope was not somehow sent to another thread.

Reply:
Done.